### ISSUE TITLE
New style attribute Value.Style.allParameters: boolean

### ISSUE THREAD

**Issue by elucash** [immutables/immutables#111]
```java
/**
 * When {@code true} &mdash; all attributes are considered as they are annotated with
 * {@link Value.Parameter}. Explicit {@link Value.Parameter} could then be used to specify
 * order, but otherwise will not make any difference, if order is not specified, source
 * definition order is used as expected. This style could be used to create special
 * tuple-style annotations:
 * 
 * <pre>
 * {@literal @}Value.Style(
 *     typeImmutable = "*Tuple",
 *     allParameters = true,
 *     defaults = {@literal @}Value.Immutable(builder = false))
 * public @interface Tuple {}
 * 
 * {@literal @}Tuple
 * {@literal @}Value.Immutable
 * interface Color {
 *   int red();
 *   int green();
 *   int blue();
 * }
 * 
 * ColorTuple.of(0xFF, 0x00, 0xFE);
 * </pre>
 * @return if all attributes will be considered parameters
 */
boolean allParameters() default false;
```

**Comment by elucash** [comment]
Tuple annotation will not be provided out of the box, but one could create it if needed

```java
@Value.Style(
    typeImmutable = "*Tuple",
    allParameters = true,
    defaults = {@literal @}Value.Immutable(builder = false))
public @interface Typle {}

@Tuple
@Value.Immutable
interface Color {
  int red();
  int green();
  int blue();
}

ColorTuple.of(0xFF, 0x00, 0xFE);
```

**Comment by jbergfi** [comment]
Great stuff, just noticed this one, this is exactly what I asked about at https://groups.google.com/forum/#!topic/immutables/JChhoTCs0SM

**Comment by elucash** [comment]
Released in 2.0.12
